pkg/controller/logstash/volume/volumes_test.go

```diff
@@ -0,0 +1,248 @@
+		{
+			name: "one es ref with ca, another no ca",
+			logstash: logstashv1alpha1.Logstash{
+				Spec: logstashv1alpha1.LogstashSpec{
+					ElasticsearchRefs: []logstashv1alpha1.ElasticsearchCluster{
+						{
+							ObjectSelector: commonv1.ObjectSelector{Name: "uat"},
+							ClusterName:    "uat",
+						},
+						{
+							ObjectSelector: commonv1.ObjectSelector{Name: "production"},
+							ClusterName:    "production",
+						},
+					},
+				},
+			},
+			setAssocConfs: func(assocs []commonv1.Association) {
+				assocs[0].SetAssociationConf(&commonv1.AssociationConf{
+					// No CASecretName
+				})
+				assocs[1].SetAssociationConf(&commonv1.AssociationConf{
+					CASecretName: "production-es-ca",
+				})
+			},
+			wantAssociationsLength: 1,
+		},
+	} {
+		t.Run(tt.name, func(t *testing.T) {
+			assocs := tt.logstash.GetAssociations()
+			tt.setAssocConfs(assocs)
+			associations, err := getVolumesFromAssociations(assocs)
+			require.NoError(t, err)
+			require.Equal(t, tt.wantAssociationsLength, len(associations))
+		})
+	}
+}
+
+func Test_BuildVolumes_DefaultVolumesMountPath(t *testing.T) {
+	hostPathType := corev1.HostPathDirectoryOrCreate
+
+	tt := []struct {
+		name     string
+		logstash logstashv1alpha1.Logstash
+		want     []corev1.ContainerPort
+	}{
+		{
+			name: "with default data PVC",
+			logstash: logstashv1alpha1.Logstash{
+				Spec: logstashv1alpha1.LogstashSpec{},
+			},
+		},
+		{
+			name: "with user provided data PVC",
+			logstash: logstashv1alpha1.Logstash{
+				Spec: logstashv1alpha1.LogstashSpec{
+					VolumeClaimTemplates: []corev1.PersistentVolumeClaim{
+						{
+							ObjectMeta: metav1.ObjectMeta{
+								Name: "logstash-data",
+							},
+							Spec: corev1.PersistentVolumeClaimSpec{
+								AccessModes: []corev1.PersistentVolumeAccessMode{
+									corev1.ReadWriteOnce,
+								},
+								Resources: corev1.ResourceRequirements{
+									Requests: corev1.ResourceList{
+										corev1.ResourceStorage: resource.MustParse("42Ti"),
+									},
+								},
+							},
+						},
+					}},
+			},
+		},
+		{
+			name: "with user provided other PVC",
+			logstash: logstashv1alpha1.Logstash{
+				Spec: logstashv1alpha1.LogstashSpec{
+					VolumeClaimTemplates: []corev1.PersistentVolumeClaim{
+						{
+							ObjectMeta: metav1.ObjectMeta{
+								Name: "pq",
+							},
+							Spec: corev1.PersistentVolumeClaimSpec{
+								AccessModes: []corev1.PersistentVolumeAccessMode{
+									corev1.ReadWriteOnce,
+								},
+								Resources: corev1.ResourceRequirements{
+									Requests: corev1.ResourceList{
+										corev1.ResourceStorage: resource.MustParse("42Ti"),
+									},
+								},
+							},
+						},
+					}},
+			},
+		},
+		{
+			name: "with user provided other PVC and logstash-data",
+			logstash: logstashv1alpha1.Logstash{
+				Spec: logstashv1alpha1.LogstashSpec{
+					VolumeClaimTemplates: []corev1.PersistentVolumeClaim{
+						{
+							ObjectMeta: metav1.ObjectMeta{
+								Name: "pq",
```

Comment:
I think the unit test can assert PVC `pq` is created and make sure the storage size of `logstash-data` is updated